scaffolding-python/lib/scaffolding.sh

```diff
@@ -11,62 +11,154 @@
 # }
 
 #
+# Plan Default Variables
+#
+pkg_bin_dirs=(${pkg_bin_dirs[@]:-"bin"})
+
+#
 # Scaffolding Variables
 #
 
-scaffolding_python_pkg="core/python"
+# This variable allows a user to set the habitat package name to be different
+# from the package name on pypi if needed.
+pkg_scaffolding_python_pip_name="${pkg_scaffolding_python_pip_name:-$pkg_name}"
+# This variable allows a user to set the habitat package version to be different
+# from the package version on pypi if needed.
+pkg_scaffolding_python_pip_version="${pkg_scaffolding_python_pip_version:-$pkg_version}"
+# Name of python package to add to the build and runtime deps.
+pkg_scaffolding_python_pkg="core/python"
+# Default venv command to use.
+pkg_scaffolding_python_venv_cmd="python -m venv"
 
 #
 # Scaffolding Callback Functions
 #
 
+# Helper function to check for a setup.py file.
+_setup_py_exists() {
+  [[ -f "$HAB_CACHE_SRC_PATH/$pkg_dirname/setup.py" ]]
+}
+
+# Initialize the variable as an empty string. This is intended for internal use
+# for this scaffolding.
+_scaffolding_python_pip_only=""
+# Helper function to disable default phases which are not used for pip installs.
+_pip_install_only() {
+  _scaffolding_python_pip_only="true"
+  pkg_interpreters=(bin/python)
+
+  do_download() {
+    return 0
+  }
+
+  do_verify() {
+    return 0
+  }
+
+  do_unpack() {
+    return 0
+  }
+}
+
 # The default_begin phase is executed prior to loading the scaffolding. As such
 # we have a scaffolding specific callback to allow us to run anything we need
 # to execute before download and build. This callback executes immediately
 # after the scaffolding is loaded.
 _scaffolding_begin() {
-  pkg_deps=(${pkg_deps[@]} $scaffolding_python_pkg)
+  pkg_deps=(${pkg_deps[@]} $pkg_scaffolding_python_pkg)
+  if _resolve_run_dependencies; then
+    build_line "Resolved Scaffolding Dependencies"
+  fi
+  # Disable download, verify, and unpack if we are only installing via pip.
+  # It should be noted that we currently must have `pkg_source` since we
+  # validate plans before scaffolding is loaded.
+  if [[ "$pkg_source" == "pip" ]] || [[ "$pkg_source" =~ ^nosuchfile.* ]] ; then
```

Comment:
There is some emergent behavior in the 0.21.0 plan-build where this can be undefined. The scaffolding currently assumes you are building the package via setup.py, but in the case of pip that is not going to be the case. I think this is the right thing to do for now, but I am open to suggestions.